src/components/interactive-builder/modals/question/question-form/common/sortable-tag/sortable-tag.component.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import { useSortable } from '@dnd-kit/sortable';
+import { CSS } from '@dnd-kit/utilities';
+import { useTranslation } from 'react-i18next';
+
+import { IconButton } from '@carbon/react';
+import { Draggable, WarningAltFilled } from '@carbon/react/icons';
+
+import styles from './sortable-tag.scss';
+
+interface SortableTagProps {
+  id: string;
+  text: string;
+  onDelete?: () => void;
+  isInvalid?: boolean;
+}
+
+export const SortableTag: React.FC<SortableTagProps> = ({ id, text, onDelete, isInvalid }) => {
+  const { t } = useTranslation();
+  const { attributes, listeners, setNodeRef, transform, transition, isDragging } = useSortable({
+    id,
+  });
+
+  const dynamicStyle = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+  };
+
+  return (
+    <div
+      ref={setNodeRef}
+      style={dynamicStyle}
+      className={`${styles.sortableTagWrapper} ${isDragging ? styles.dragContainerWhenDragging : ''}`}
+    >
+      <div className={styles.leftContent}>
+        <div {...attributes} {...listeners} className={styles.dragHandle}>
+          <IconButton className={styles.dragIcon} kind="ghost" size="md" label={t('dragToReorder', 'Drag to reorder')}>
+            <Draggable />
+          </IconButton>
+        </div>
+        <span
+          className={`${styles.sortableTag} ${isDragging ? styles.dragging : ''} ${isInvalid ? styles.invalid : ''}`}
+        >
+          {text}
+        </span>
+      </div>
+      <div>
+        {onDelete ? (
+          <button
+            aria-label="Clear all selected items"
+            onClick={(e) => {
+              e.stopPropagation();
+              onDelete();
+            }}
+            className={styles.conceptAnswerButton}
+          >
+            X
+          </button>
+        ) : (
+          isInvalid && <WarningAltFilled className={styles.invalidIcon} aria-label="Invalid answer" size={16} />
+        )}
```

Comment:
A couple of things here:
1. In the future, please always use Carbon elements over pure html elements
2. In this case, you can use Carbon's `IconButton`
3. Since if `isInvalid` is true, the invalid icon should be shown, it should not be the else condition to if the `onDelete` function is present
4. The `invalidIcon` style doesn't exist in the stylesheet
```suggestion
 {onDelete && (
          <IconButton
            enterDelayMs={300}
            label={t('deleteAnswer', 'Delete answer')}
            kind="ghost"
            onClick={onDelete}
            size="md"
          >
            <TrashCan />
          </IconButton>
        )}
        {isInvalid && <WarningAltFilled className={styles.invalidIcon} aria-label="Invalid answer" size={16} />}
```